File: backend/Application/Transacoes/ListTransacoes/ListTransacoesResponse.cs
namespace FinancialControllerServer.Application.Transacoes.ListTransacoes;

public class ListTransacoesResponse
{
    public Guid Id { get; set; }
    public required string Descricao { get; set; }
    public decimal Valor { get; set; }
    public int Tipo { get; set; }
    public DateOnly Data { get; set; }
    public required string PessoaNome { get; set; }
    public required string CategoriaDescricao { get; set; }
}

File: frontend/src/features/transacoes/types/listTransacoesResponse.ts
import type { TipoTransacao } from "../../../domain/enums/TipoTransacao";

export interface ListTransacoesResponse {
    id: string,
    descricao: string;
    valor: number;
    tipo: TipoTransacao;
    data: string;
    pessoaNome: string;
    categoriaDescricao: string;
}